ipod-shuffle-4g.py

```diff
@@ -14,15 +14,36 @@
 import re
 import tempfile
 import signal
+import enum
+import functools
 
 # External libraries
 try:
     import mutagen
 except ImportError:
     mutagen = None
 
-audio_ext = (".mp3", ".m4a", ".m4b", ".m4p", ".aa", ".wav")
-list_ext = (".pls", ".m3u")
+class PlaylistType(enum.Enum):
+    ALL_SONGS = 1
+    NORMAL = 2
+    PODCAST = 3
+    AUDIOBOOK = 4
+
+class FileType(enum.Enum):
+    MP3 = (1, {'.mp3'})
+    AAC = (2, {'.m4a', '.m4b', '.m4p', '.aa'})
+    WAV = (4, {'.wav'})
+    def __init__(self, filetype, extensions):
+        self.filetype = filetype
+        self.extensions = extensions
+
+# collect all the supported audio extensions
+audio_ext = functools.reduce(lambda j,k: j.union(k), map(lambda i: i.extensions, FileType))
```

Comment:
Doesnt it make sense to include that inside the `FileType` Class?

Reply:
Maybe I'm missing how to do this, but my attempts haven't succeeded and the examples I have seen for defining this as a static attribute involve extra lines. Also, given that it's right above two other lists of extensions, I don't think it's objectionable to have it outside the class.